x-pack/plugins/cases/public/containers/api.ts

```diff
@@ -182,9 +183,18 @@ export const getCases = async ({
   const query = {
     ...(filterOptions.status !== StatusAll ? { status: filterOptions.status } : {}),
     ...(filterOptions.severity !== SeverityAll ? { severity: filterOptions.severity } : {}),
-    assignees: filterOptions.assignees,
-    reporters: constructReportersFilter(filterOptions.reporters),
-    tags: filterOptions.tags,
+    ...(filterOptions.assignees === null || filterOptions.assignees.length > 0
+      ? {
+          assignees:
+            filterOptions.assignees?.map((assignee) =>
+              assignee === null ? NO_ASSIGNEES_FILTERING_KEYWORD : assignee
+            ) ?? NO_ASSIGNEES_FILTERING_KEYWORD,
+        }
+      : {}),
+    ...(filterOptions.reporters.length > 0
```

Comment:
Let's move the entire logic for reporters inside of the `constructReportersFilter` so it returns an object that can be spread.